be/src/exec/vectorized/orc_scanner_adapter.cpp

```diff
@@ -52,6 +52,14 @@ const static std::set<PrimitiveType> g_starrocks_decimal_type = {TYPE_DECIMAL32,
 const static cctz::time_point<cctz::sys_seconds> CCTZ_UNIX_EPOCH =
         std::chrono::time_point_cast<cctz::sys_seconds>(std::chrono::system_clock::from_time_t(0));
 
+// Hive ORC char type will pad trailing spaces.
+// https://docs.cloudera.com/documentation/enterprise/6/6.3/topics/impala_char.html
+static inline size_t remove_trailing_spaces(const char* s, size_t size) {
+    size -= 1;
+    while (size >= 0 && s[size] == ' ') size--;
```

Comment:
size_t was uint64_t, so size >=0 is always true

Reply:
nice. catch. thank you.